nimare/meta/cbma/base.py

```diff
@@ -0,0 +1,476 @@
+"""
+CBMA methods from the ALE and MKDA families.
+"""
+import logging
+import multiprocessing as mp
+import inspect
+
+import nibabel as nib
+import numpy as np
+import pandas as pd
+from scipy import ndimage
+from tqdm.auto import tqdm
+
+from ...base import MetaEstimator
+from ...results import MetaResult
+from ...stats import null_to_p
+from ...transforms import p_to_z
+from ...utils import round2
+
+LGR = logging.getLogger(__name__)
+
+
+class CBMAEstimator(MetaEstimator):
```

Comment:
This probably should happen now, but I'm thinking that we may want to further divide this into a `DensityBasedEstimator` and something else, so separate density-based approaches like ALE and MKDA from dramatically different ones like SDM-PSI.

Reply:
Agreed. I initially wanted to do it that way (i.e., to put all the abstracted methods into a `DensityCBMAEstimator` class), but figured we should see how far we can get by consolidating everything, instead of multiplying classes in advance. If we can't re-use everything in there right now for the pairwise estimators, we may also need to define separate `OneSample` and `Pairwise` subclasses.